File: 프로그래머스/python/LV2/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    dic = {}
    for i in tangerine:
        if i in dic:
            dic[i] += 1
        else:
            dic[i] = 1
    sorted_dic = dict(sorted(dic.items(), key=lambda x: -x[1]))

    for i in sorted_dic:
        if k <= 0:
            continue
        k -= sorted_dic[i]
        answer += 1
    return answer

File: 프로그래머스/javascript/LV2/귤 고르기.js
function solution(k, tangerine) {
    let answer = 0;
    const obj = {};
    
    for(let i=0;i<tangerine.length;i++){
        if(obj[tangerine[i]]){
            obj[tangerine[i]]++;
        }else{
            obj[tangerine[i]] = 1;
        }        
    }
    
    const sorted = Object.entries(obj).sort((a,b)=>b[1]-a[1]);
    
    for(let i=0;i<sorted.length;i++){
        answer++;
        if(k>sorted[i][1]){
            k -= sorted[i][1];
        }else{
            break;
        }
    }
    return answer;
}

